test/nls_consistency.jl

```diff
@@ -23,77 +26,85 @@ function consistent_nls_functions(nlss; nloops=10, rtol=1.0e-8)
   for k = 1:nloops
     x = 10 .* (rand(n) .- 0.5)
 
-    Fs = Any[residual(nls, x) for nls in nlss]
-    for i = 1:N
-      for j = i+1:N
-        @test isapprox(Fs[i], Fs[j], rtol=rtol)
-      end
+    if !(residual in exclude)
+      Fs = Any[residual(nls, x) for nls in nlss]
+      for i = 1:N
+        for j = i+1:N
+          @test isapprox(Fs[i], Fs[j], rtol=rtol)
+        end
 
-      r = residual!(nlss[i], x, tmp_m)
-      @test isapprox(r, Fs[i], rtol=rtol)
-      @test isapprox(Fs[i], tmp_m, rtol=rtol)
+        r = residual!(nlss[i], x, tmp_m)
+        @test isapprox(r, Fs[i], rtol=rtol)
+        @test isapprox(Fs[i], tmp_m, rtol=rtol)
+      end
     end
 
-    Js = Any[jac_residual(nls, x) for nls in nlss]
-    for i = 1:N
-      for j = i+1:N
-        @test isapprox(Js[i], Js[j], rtol=rtol)
+    if !(jac_residual in exclude)
+      Js = Any[jac_residual(nls, x) for nls in nlss]
+      for i = 1:N
+        for j = i+1:N
+          @test isapprox(Js[i], Js[j], rtol=rtol)
+        end
       end
     end
 
-    J_ops = Any[jac_op_residual(nls, x) for nls in nlss]
-    Jv, Jtv = zeros(m), zeros(n)
-    J_ops_inplace = Any[jac_op_residual!(nls, x, Jv, Jtv) for nls in nlss]
+    if intersect([jac_op_residual, jprod_residual, jtprod_residual],  exclude) == []
+      J_ops = Any[jac_op_residual(nls, x) for nls in nlss]
+      Jv, Jtv = zeros(m), zeros(n)
+      J_ops_inplace = Any[jac_op_residual!(nls, x, Jv, Jtv) for nls in nlss]
 
-    v = rand(n)
+      v = rand(n)
```

Comment:
There's still a random vector here, and another one below. I would replace them with [1, -1, 1, -1, ...].

Reply:
I would prefer some random instance(s) for example using one or several seeds to have reproducible results. A simple test with say [1, -1, ...] may fail to exhibit issues because of its evident symmetry and specificity.

Reply:
Sorry, missed those because of `git diff -w`.

Reply:
I don't think a seed will guarantee reproducible results on various platforms. The vector [1, -1, 1, -1, ...] is often used in place of a random vector and will give reproducible results. Yes it may fail to exhibit some issues, but it will not give false positives.